pkg/chunk/cassandra/storage_client_test.go

```diff
@@ -0,0 +1,80 @@
+package cassandra
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gocql/gocql"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfig_setClusterConfig_noAuth(t *testing.T) {
+	cqlCfg := gocql.NewCluster()
+	cfg := Config{
+		Auth: false,
+	}
+	require.NoError(t, cfg.Validate())
+
+	err := cfg.setClusterConfig(cqlCfg)
+	require.NoError(t, err)
+	assert.Nil(t, cqlCfg.Authenticator)
+}
+
+func TestConfig_setClusterConfig_authWithPassword(t *testing.T) {
+	cqlCfg := gocql.NewCluster()
+	cfg := Config{
+		Auth:     true,
+		Username: "user",
+		Password: "pass",
+	}
+	require.NoError(t, cfg.Validate())
+
+	err := cfg.setClusterConfig(cqlCfg)
+	require.NoError(t, err)
+	assert.NotNil(t, cqlCfg.Authenticator)
+	assert.Equal(t, "user", cqlCfg.Authenticator.(gocql.PasswordAuthenticator).Username)
+	assert.Equal(t, "pass", cqlCfg.Authenticator.(gocql.PasswordAuthenticator).Password)
+}
+
+func TestConfig_setClusterConfig_authWithPasswordFile_noTailLn(t *testing.T) {
+	cqlCfg := gocql.NewCluster()
+	cfg := Config{
+		Auth:         true,
+		Username:     "user",
+		PasswordFile: "testdata/pass1.txt",
```

Comment:
I would suggest to rename this file to better explain what it should contain (ie. `password-without-trailing-newline.txt`)